### PR TITLE
Add groups support to window functions

### PR DESCRIPTION
**Pull request by feilong-liu**
Cherry-pick of https://github.com/trinodb/trino/pull/5713

Co-authored-by: kasiafi <30203062+kasiafi@users.noreply.github.com>

Test plan - (Please fill in how you tested your changes)

Unit tests

```
== RELEASE NOTES ==

General Changes
* Add groups support to window function.
```

### CODE REVIEW

File: presto-docs/src/main/sphinx/functions/window.rst [hunk 1]
```diff
@@ -58,6 +58,10 @@ The window definition has 3 components:
   the current row. The sort key must either be the same type of ``expression`` or can be coerced to the
   same type as ``expression``.
 
+  In ``GROUPS`` mode, frame starts and ends of ``expression PRECEDING`` or ``expression FOLLOWING``
```
Comment (pranjalssh) [inline presto-docs/src/main/sphinx/functions/window.rst:61]: Its not clear what a group represents. Can you explain what a group means? A query example below will also be good to add

Comment (feilong-liu) [reply presto-docs/src/main/sphinx/functions/window.rst:61]: Add explanation and also examples to demonstrate the three types of frame definitions.

File: presto-docs/src/main/sphinx/functions/window.rst [hunk 2]
```diff
@@ -58,6 +58,11 @@ The window definition has 3 components:
...
+  define the start or end of the frame as the number of groups from the current row.
+  Here a group includes all rows with the same value on the sorting key.
```
Comment (pranjalssh) [inline presto-docs/src/main/sphinx/functions/window.rst:63]:
```suggestion
  A group includes all rows with the same value on the sorting key.
```

File: presto-docs/src/main/sphinx/functions/window.rst [hunk 3]
```diff
@@ -71,6 +76,50 @@ The following query ranks orders for each clerk by price::
...
 
+The following queries demonstrate the difference between ``ROWS``, ``RANGE`` and ``GROUPS`` in frame definition::
+
+    SELECT
+        ARRAY_AGG(v) OVER (
+            ORDER BY
+                k ASC ROWS BETWEEN 1 PRECEDING AND 1 following
+        )
+    FROM (
+        VALUES
```
Comment (pranjalssh) [inline presto-docs/src/main/sphinx/functions/window.rst:87]: Is it possible to put VALUES in 1 line, so documentation doesn't look too long?

**Review by pranjalssh — APPROVED** [review]

**Pull request merged**